src/main/java/org/jabref/model/search/rules/RegexBasedSearchRule.java

```diff
@@ -92,7 +90,11 @@ public PdfSearchResults getFulltextResults(String query, BibEntry bibEntry) {
             this.lastQuery = query;
             lastSearchResults = List.of();
             try {
-                PdfSearcher searcher = PdfSearcher.of(databaseContext);
+                if (Globals.stateManager.getActiveDatabase().isEmpty()) {
```

Comment:
Although we agreed to use the Globals here, I would  nonetheless add it to the constructor 
e.g. this.stateManager = Globals.stateManager so it's easier to spot for future refactorings and not  hidden somehwere in a method down below

Reply:
Done in 85beb5c